src/main/Application.h

```diff
@@ -175,6 +226,11 @@ class Application
     virtual std::string getStateHuman() const = 0;
     virtual bool isStopping() const = 0;
 
+    // Gets the current Application mode. This is a constant established
+    // when the Application is constructed and will not change while it
+    // runs.
+    virtual AppMode getMode() const = 0;
```

Comment:
hm, we have such interfaces as static helpers (Application::modeHasFoo(AppMode)); I figured it might be helpful to have the flexibility to switch or use the helpers in different contexts (and/or reify the mode as a value you might want to store / pass around). I guess these are hypotheticals though; can change if you feel strongly!

Reply:
Yeah I agree, the code is littered with `Application::modeHasDatabase(mApp.getMode())` . 
Also, something like `mApp.hasDatabase()` is easier to read

Reply:
fixed in all cases (though delegating to the static methods, which are still there for querying a mode _before_ the application is built).